### ISSUE TITLE
Model `db.backup` does not exist.

### ISSUE THREAD

**Issue by Yenthe666** [Yenthe666/auto_backup#65]
Some users report that they get a traceback / error saying that the db.backup model is not found.
This issue also seems to be possible if you have the module already installed and seems to happen somewhere randomly in time. Example from one of my production logs:
```
	Line 325566: 2017-08-20 18:45:26,657 7423 INFO mydb openerp.addons.auto_backup.backup_scheduler: Copying File /odoo/backups/20_08_2017_18_00_20_mydb.zip------ success
	Line 373775: 2017-08-21 18:00:03,605 2609 WARNING mydb openerp.addons.base.ir.ir_cron: Model `db.backup` does not exist.
	Line 418334: 2017-08-22 18:00:11,203 2609 WARNING mydb openerp.addons.base.ir.ir_cron: Model `db.backup` does not exist.
	Line 462467: 2017-08-23 18:00:23,201 2609 WARNING mydb openerp.addons.base.ir.ir_cron: Model `db.backup` does not exist.
	Line 526564: 2017-08-24 18:00:32,859 2609 WARNING mydb openerp.addons.base.ir.ir_cron: Model `db.backup` does not exist
```

As you can see this in this logfile the backup worked fine the 20 augustus and since the 21'st of august is spits out `Model db.backup does not exist.` for some reason.

**Comment by Yenthe666** [comment]
For those who land here and are working on Odoo 9.. Please try this branch of the module: https://github.com/Yenthe666/auto_backup/tree/9.0-security-configuration and report back. This should probably fix the issues..

**Comment by NeatNerdPrime** [comment]
Hello @Yenthe666 , I'd like to comment on this issue and offer my solution to this. 

TL;DR: use "sudo easy_install pysftp" instead of pip

for days i was grinding my gears on why it would not work! And i was absolutely sure pysftp was properly installed as per your documentation. Still the odoo logs were complaining that  pysftp was not installed The it hit me... "what if not all dependencies of pysftp were actually met?" Because the only reason it would raise that error is because the module existed, but the dependencies were not.

You can install python modules in several ways, directly, with pip or with easy_install. 
Installation with pip does install the pysftp package, but none of it's required dependencies. I've concluded this by running a python session and trying to import the module manually. It kept complaining about missing module bcrypt, missing module nacl.signing, and other modules i didnt even knew they existed.

So how to install all these deps? Then i remembered there was such a thing as "easy_install" that also pulls in the required dependencies for the module and compiles everything. Sure it makes install time a lot longer, but in the end, your module installs properly without the error of "model 'db.backup' does not exist'.

I've reproduced this on a clean bootstrapped machine with the easy_install instead of pip ... and voila! Bazinga!

If you could be so kind to reproduce this theory, and i'm pretty sure you would get the correct results.

**Comment by NeatNerdPrime** [comment]
@Yenthe666  A PR is made for this purpose, would you be so kind to review it?: https://github.com/Yenthe666/auto_backup/pull/76

**Comment by Yenthe666** [comment]
Hi @NeatNerdPrime,

That is actually an interesting concept/fact. I've tried this on a clean VM that had no dependencies installed yet and it does look to handle all associated dependencies. My output:
```
administrator@ubuntu:~$ sudo easy_install pysftp
[sudo] password for administrator: 
Searching for pysftp
Reading https://pypi.python.org/simple/pysftp/
Best match: pysftp 0.2.9
Downloading https://pypi.python.org/packages/36/60/45f30390a38b1f92e0a8cf4de178cd7c2bc3f874c85430e40ccf99df8fe7/pysftp-0.2.9.tar.gz#md5=abc55b7122de3e86f0d547301f4ddf0c
Processing pysftp-0.2.9.tar.gz
Writing /tmp/easy_install-G2YIkP/pysftp-0.2.9/setup.cfg
Running pysftp-0.2.9/setup.py -q bdist_egg --dist-dir /tmp/easy_install-G2YIkP/pysftp-0.2.9/egg-dist-tmp-bl5Mqv
no previously-included directories found matching 'docs/_build'
zip_safe flag not set; analyzing archive contents...
Moving pysftp-0.2.9-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Adding pysftp 0.2.9 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/pysftp-0.2.9-py2.7.egg
Processing dependencies for pysftp
Searching for paramiko>=1.17
Reading https://pypi.python.org/simple/paramiko/
Best match: paramiko 2.4.0
Downloading https://pypi.python.org/packages/c8/de/791773d6a4b23327c7475ae3d7ada0d07fa147bf77fb6f561a4a7d8afd11/paramiko-2.4.0.tar.gz#md5=ed6d085a5e5369817e49cf1315b32eba
Processing paramiko-2.4.0.tar.gz
Writing /tmp/easy_install-OSO5_D/paramiko-2.4.0/setup.cfg
Running paramiko-2.4.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-OSO5_D/paramiko-2.4.0/egg-dist-tmp-CNN4MX
zip_safe flag not set; analyzing archive contents...
Moving paramiko-2.4.0-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Adding paramiko 2.4.0 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/paramiko-2.4.0-py2.7.egg
Searching for pynacl>=1.0.1
Reading https://pypi.python.org/simple/pynacl/
Best match: PyNaCl 1.2.0
Downloading https://pypi.python.org/packages/fa/38/0f35ec4beb6562f1abfa07914db1cea978e93da409ba6293f810d9e677d6/PyNaCl-1.2.0.tar.gz#md5=7a31e95f5f457184b1660b71d6808b19
Processing PyNaCl-1.2.0.tar.gz
Writing /tmp/easy_install-EDGagr/PyNaCl-1.2.0/setup.cfg
Running PyNaCl-1.2.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-EDGagr/PyNaCl-1.2.0/egg-dist-tmp-zh8mQ2
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libffi' found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libffi' found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libffi' found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libffi' found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libffi' found
c/_cffi_backend.c:15:17: fatal error: ffi.h: No such file or directory
compilation terminated.
error: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```

Funny enough this way of installing tends to break on one of those child dependencies.  I'll wait a few days to see if this gets fixed in the dependency before testing again.
P.S: Thanks a lot for the detailed info and feedback!

**Comment by jpaulin** [comment]
@Yenthe666 @NeatNerdPrime FYI:
I bumped into the same issue. Ie. my goal is to get the auto_backup module installed. Having a local Odoo 10 running on this machine. Thought also that I had previously satisfied the pysftp package dependency. Looking forward! I'll report anything new that comes up.  

UPDATE:
**The mentioned easy_install method fixed the error on my host**
(ie. sudo easy_install pysftp )

This is a Ubuntu 16.x and having Odoo 10 locally installed.

**Comment by HomaySoft** [comment]
Hello

Trying to install this module on v10 enterprise we received the same error "model not found db.backup". The module did not install. Before installing the module, we installed pysftp through "sudo pip install pysftp" and then "sudo easy_install pysftp" . The v10 enterprise is the standard and downloaded the module from apps.odoo.com for v10. Has there been any fix for this for v10?

Find below error output:

Apps
App Store
Updates
Update Apps List
Apply Scheduled Upgrades
Import Module
 
سیما 97 

 Administrator (FirstDB) 
Apps
Module
backup

Search...

Database Auto-Backup
Backups
auto_backup

×
Odoo
Odoo Server Error
Traceback (most recent call last):
  File "/home/odoo10e/odoo/odoo/http.py", line 638, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo10e/odoo/odoo/http.py", line 675, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo10e/odoo/odoo/http.py", line 331, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo10e/odoo/odoo/service/model.py", line 119, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo10e/odoo/odoo/http.py", line 324, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo10e/odoo/odoo/http.py", line 933, in __call__
    return self.method(*args, **kw)
  File "/home/odoo10e/odoo/odoo/http.py", line 504, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo10e/odoo/addons/web/controllers/main.py", line 882, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo10e/odoo/addons/web/controllers/main.py", line 870, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo10e/odoo/odoo/api.py", line 681, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/odoo10e/odoo/odoo/api.py", line 672, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo10e/odoo/odoo/addons/base/module/module.py", line 410, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/home/odoo10e/odoo/odoo/addons/base/module/module.py", line 484, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/home/odoo10e/odoo/odoo/modules/registry.py", line 78, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/odoo10e/odoo/odoo/modules/loading.py", line 339, in load_modules
    loaded_modules, update_module)
  File "/home/odoo10e/odoo/odoo/modules/loading.py", line 237, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/home/odoo10e/odoo/odoo/modules/loading.py", line 156, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/home/odoo10e/odoo/odoo/modules/loading.py", line 95, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/home/odoo10e/odoo/odoo/tools/convert.py", line 848, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/home/odoo10e/odoo/odoo/tools/convert.py", line 918, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/home/odoo10e/odoo/odoo/tools/convert.py", line 799, in parse
    self.parse(rec, mode)
  File "/home/odoo10e/odoo/odoo/tools/convert.py", line 802, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/home/odoo10e/odoo/odoo/tools/convert.py", line 710, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/home/odoo10e/odoo/odoo/addons/base/ir/ir_model.py", line 1196, in _update
    record = record.create(values)
  File "/home/odoo10e/enterprise/web_studio/models/studio_mixin.py", line 18, in create
    res = super(StudioMixin, self).create(vals)
  File "/home/odoo10e/odoo/odoo/addons/base/ir/ir_ui_view.py", line 360, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/home/odoo10e/odoo/odoo/models.py", line 3829, in create
    self._fields[key].determine_inverse(record)
  File "/home/odoo10e/odoo/odoo/fields.py", line 999, in determine_inverse
    getattr(records, self.inverse)()
  File "/home/odoo10e/odoo/odoo/addons/base/ir/ir_ui_view.py", line 220, in _inverse_arch
    view.write(data)
  File "/home/odoo10e/enterprise/web_studio/models/studio_mixin.py", line 27, in write
    res = super(StudioMixin, self).write(vals)
  File "/home/odoo10e/odoo/odoo/addons/base/ir/ir_ui_view.py", line 378, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/home/odoo10e/odoo/odoo/models.py", line 3555, in write
    self._write(old_vals)
  File "/home/odoo10e/odoo/odoo/models.py", line 3706, in _write
    self._validate_fields(vals)
  File "/home/odoo10e/odoo/odoo/models.py", line 1074, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
ParseError: "Error while validating constraint

Model not found: db.backup

Error context:
View `Configure Backup`
[view_id: 1457, xml_id: n/a, model: db.backup, parent_id: n/a]
None" while parsing /home/odoo10e/odoo/addons/auto_backup/views/bkp_conf_view.xml:4, near
<record model="ir.ui.view" id="view_backup_conf_form">
      <field name="name">Configure Backup</field>
      <field name="model">db.backup</field>
      <field name="type">form</field>
      <field name="arch" type="xml">
        <form string="Test">
	  <group col="4" colspan="4">
	    <separator col="2" string="Local backup configuration"/>
          </group>
	  <group>
	    <field name="host" colspan="2"/>
	    <field name="name"/>
	    <field name="port"/>
	    <field name="backup_type"/>
	    <field name="folder"/>
	    <field name="autoremove"/>
	    <field name="days_to_keep" attrs="{'invisible': [('autoremove','=',False)]}"/>
	  </group>
	  <group col="4" colspan="4">
	    <separator col="2" string="SFTP"/>
          </group>
          <div style="width:50%;border-radius:10px;margin: 10px 0px;padding:15px 10px 15px 10px; background-repeat: no-repeat;background-position: 10px center;color: #9F6000;background-color: #FEEFB3;" attrs="{'invisible': [('sftp_write','=',False)]}"><b>Warning:</b> Use SFTP with caution! This writes files to external servers under the path you specify.</div>
	    <group>
              <field name="sftp_write"/>
              <field name="sftp_host" attrs="{'invisible':[('sftp_write', '=', False)],'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_port" attrs="{'invisible':[('sftp_write', '=', False)],'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_user" attrs="{'invisible':[('sftp_write', '=', False)], 'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_password" attrs="{'invisible':[('sftp_write', '=', False)],'required': [('sftp_write', '=', True)]}" password="True"/>
              <field name="sftp_path" attrs="{'invisible':[('sftp_write', '==', False)],'required':[('sftp_write', '==', True)]}" placeholder="For example: /odoo/backups/"/>
              <field name="days_to_keep_sftp" attrs="{'invisible': [('sftp_write', '=', False)], 'required': [('sftp_write', '=', True)]}"/>
              <field name="send_mail_sftp_fail" attrs="{'invisible': [('sftp_write','=',False)]}"/>
	      <field name="email_to_notify" attrs="{'invisible':['|',('send_mail_sftp_fail', '==', False), ('sftp_write', '=', False)], 'required': [('send_mail_sftp_fail', '=', True)]}"/>
              <button name="test_sftp_connection" type="object" attrs="{'invisible': [('sftp_write','=',False)]}" string="Test SFTP Connection" icon="gtk-network"/>
            </group>
            <separator string="Help" colspan="2"/>
            <div>
              This configures the scheduler for automatic backup of the given database running on given host at given port on regular intervals.
              <br/>
	      Automatic backups of the database can be scheduled as follows: 
	      <ol>
		<li>
		  Go to Settings / Technical / Automation / Scheduled Actions.
		</li>
		<li>
		  Search the action named 'Backup scheduler'.
		</li>
		<li>
		  Set the scheduler to active and fill in how often you want backups generated.
		</li>
              </ol>
	      <p style="font-size:18px;">
		Need more help? <a href="mailto:yenthe.vanginneken@vanroey.be">Contact us!</a>
	      </p>
	  </div>
        </form>
      </field>
    </record>

**Comment by morrillo** [comment]
I got the same error, "Model db.backup does not exist". The problem is the module does not install due to problems with pyOpenSSL. This python dependency is either outdated or not installed. Can be installed with

pip install pyOpenSSL --upgrade

**Comment by Yenthe666** [comment]
Should have been fixed with the requirements changes and future updates.

**Comment by prajithpipt** [comment]
@Yenthe666 

I have the same problem 

I downloaded the auto_backup module and try to install in odoo 12. But  can't install due to the error message mentioned below 

  File "/home/ksoftodoo/odoo12/odoo/models.py", line 3569, in create
    fields[0].determine_inverse(batch_recs)
  File "/home/ksoftodoo/odoo12/odoo/fields.py", line 1114, in determine_inverse
    getattr(records, self.inverse)()
  File "/home/ksoftodoo/odoo12/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/home/ksoftodoo/odoo12/addons/website/models/ir_ui_view.py", line 41, in write
    return super(View, self).write(vals)
  File "/home/ksoftodoo/odoo12/odoo/addons/base/models/ir_ui_view.py", line 444, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/home/ksoftodoo/odoo12/odoo/models.py", line 3284, in write
    self._write(store_vals)
  File "/home/ksoftodoo/odoo12/odoo/models.py", line 3430, in _write
    self._validate_fields(vals)
  File "/home/ksoftodoo/odoo12/odoo/models.py", line 1112, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Model not found: db.backup

Error context:
View `Configure Backup`
[view_id: 3007, xml_id: n/a, model: db.backup, parent_id: n/a]
None" while parsing /home/ksoftodoo/odoo12/custom_addons/auto_backup-10.0.10.0/auto_backup/views/bkp_conf_view.xml:4, near
<record model="ir.ui.view" id="view_backup_conf_form">
      <field name="name">Configure Backup</field>
      <field name="model">db.backup</field>
      <field name="type">form</field>
      <field name="arch" type="xml">
        <form string="Test">
	  <group col="4" colspan="4">
	    <separator col="2" string="Local backup configuration"/>
          </group>
	  <group>
	    <field name="host" colspan="2"/>
	    <field name="name"/>
	    <field name="port"/>
	    <field name="backup_type"/>
	    <field name="folder"/>
	    <field name="autoremove"/>
	    <field name="days_to_keep" attrs="{'invisible': [('autoremove','=',False)]}"/>
	  </group>
	  <group col="4" colspan="4">
	    <separator col="2" string="SFTP"/>
          </group>
          <div style="width:50%;border-radius:10px;margin: 10px 0px;padding:15px 10px 15px 10px; background-repeat: no-repeat;background-position: 10px center;color: #9F6000;background-color: #FEEFB3;" attrs="{'invisible': [('sftp_write','=',False)]}"><b>Warning:</b> Use SFTP with caution! This writes files to external servers under the path you specify.</div>
	    <group>
              <field name="sftp_write"/>
              <field name="sftp_host" attrs="{'invisible':[('sftp_write', '=', False)],'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_port" attrs="{'invisible':[('sftp_write', '=', False)],'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_user" attrs="{'invisible':[('sftp_write', '=', False)], 'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_password" attrs="{'invisible':[('sftp_write', '=', False)],'required': [('sftp_write', '=', True)]}" password="True"/>
              <field name="sftp_path" attrs="{'invisible':[('sftp_write', '==', False)],'required':[('sftp_write', '==', True)]}" placeholder="For example: /odoo/backups/"/>
              <field name="days_to_keep_sftp" attrs="{'invisible': [('sftp_write', '=', False)], 'required': [('sftp_write', '=', True)]}"/>
              <field name="send_mail_sftp_fail" attrs="{'invisible': [('sftp_write','=',False)]}"/>
	      <field name="email_to_notify" attrs="{'invisible':['|',('send_mail_sftp_fail', '==', False), ('sftp_write', '=', False)], 'required': [('send_mail_sftp_fail', '=', True)]}"/>
              <button name="test_sftp_connection" type="object" attrs="{'invisible': [('sftp_write','=',False)]}" string="Test SFTP Connection" icon="gtk-network"/>
            </group>
            <separator string="Help" colspan="2"/>
            <div>
              This configures the scheduler for automatic backup of the given database running on given host at given port on regular intervals.
              <br/>
	      Automatic backups of the database can be scheduled as follows: 
	      <ol>
		<li>
		  Go to Settings / Technical / Automation / Scheduled Actions.
		</li>
		<li>
		  Search the action named 'Backup scheduler'.
		</li>
		<li>
		  Set the scheduler to active and fill in how often you want backups generated.
		</li>
              </ol>
	      <p style="font-size:18px;">
		Need more help? <a href="mailto:yenthe.vanginneken@vanroey.be">Contact us!</a>
	      </p>
	  </div>
        </form>
      </field>
    </record>

**Comment by prajithpipt** [comment]
@Yenthe666 

I have the same issue during the installation of the 'auto_backup' module in 'odoo 12'

 File "/home/ksoftodoo/odoo12/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/home/ksoftodoo/odoo12/addons/website/models/ir_ui_view.py", line 41, in write
    return super(View, self).write(vals)
  File "/home/ksoftodoo/odoo12/odoo/addons/base/models/ir_ui_view.py", line 444, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/home/ksoftodoo/odoo12/odoo/models.py", line 3284, in write
    self._write(store_vals)
  File "/home/ksoftodoo/odoo12/odoo/models.py", line 3430, in _write
    self._validate_fields(vals)
  File "/home/ksoftodoo/odoo12/odoo/models.py", line 1112, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Model not found: db.backup

Error context:
View `Configure Backup`
[view_id: 3010, xml_id: n/a, model: db.backup, parent_id: n/a]
None" while parsing /home/ksoftodoo/odoo12/custom_addons/auto_backup-10.0.10.0/auto_backup/views/bkp_conf_view.xml:4, near
<record model="ir.ui.view" id="view_backup_conf_form">
      <field name="name">Configure Backup</field>
      <field name="model">db.backup</field>
      <field name="type">form</field>
      <field name="arch" type="xml">
        <form string="Test">
	  <group col="4" colspan="4">
	    <separator col="2" string="Local backup configuration"/>
          </group>
	  <group>
	    <field name="host" colspan="2"/>
	    <field name="name"/>
	    <field name="port"/>
	    <field name="backup_type"/>
	    <field name="folder"/>
	    <field name="autoremove"/>
	    <field name="days_to_keep" attrs="{'invisible': [('autoremove','=',False)]}"/>
	  </group>
	  <group col="4" colspan="4">
	    <separator col="2" string="SFTP"/>
          </group>
          <div style="width:50%;border-radius:10px;margin: 10px 0px;padding:15px 10px 15px 10px; background-repeat: no-repeat;background-position: 10px center;color: #9F6000;background-color: #FEEFB3;" attrs="{'invisible': [('sftp_write','=',False)]}"><b>Warning:</b> Use SFTP with caution! This writes files to external servers under the path you specify.</div>
	    <group>
              <field name="sftp_write"/>
              <field name="sftp_host" attrs="{'invisible':[('sftp_write', '=', False)],'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_port" attrs="{'invisible':[('sftp_write', '=', False)],'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_user" attrs="{'invisible':[('sftp_write', '=', False)], 'required':[('sftp_write', '=', True)]}"/>
              <field name="sftp_password" attrs="{'invisible':[('sftp_write', '=', False)],'required': [('sftp_write', '=', True)]}" password="True"/>
              <field name="sftp_path" attrs="{'invisible':[('sftp_write', '==', False)],'required':[('sftp_write', '==', True)]}" placeholder="For example: /odoo/backups/"/>
              <field name="days_to_keep_sftp" attrs="{'invisible': [('sftp_write', '=', False)], 'required': [('sftp_write', '=', True)]}"/>
              <field name="send_mail_sftp_fail" attrs="{'invisible': [('sftp_write','=',False)]}"/>
	      <field name="email_to_notify" attrs="{'invisible':['|',('send_mail_sftp_fail', '==', False), ('sftp_write', '=', False)], 'required': [('send_mail_sftp_fail', '=', True)]}"/>
              <button name="test_sftp_connection" type="object" attrs="{'invisible': [('sftp_write','=',False)]}" string="Test SFTP Connection" icon="gtk-network"/>
            </group>
            <separator string="Help" colspan="2"/>
            <div>
              This configures the scheduler for automatic backup of the given database running on given host at given port on regular intervals.
              <br/>
	      Automatic backups of the database can be scheduled as follows: 
	      <ol>
		<li>
		  Go to Settings / Technical / Automation / Scheduled Actions.
		</li>
		<li>
		  Search the action named 'Backup scheduler'.
		</li>
		<li>
		  Set the scheduler to active and fill in how often you want backups generated.
		</li>
              </ol>
	      <p style="font-size:18px;">
		Need more help? <a href="mailto:yenthe.vanginneken@vanroey.be">Contact us!</a>
	      </p>
	  </div>
        </form>
      </field>
    </record>

**Comment by prajithpipt** [comment]
@Yenthe666 
auto_backup Installation Error in Odoo 12

from . import db_backup
  File "/home/ksoftodoo/odoo12/custom_addons/auto_backup-12.0/auto_backup/models/db_backup.py", line 26, in <module>
    raise ImportError('This module needs paramiko to automatically write backups to the FTP through SFTP. Please install paramiko on your system. (sudo pip3 install paramiko)')
ImportError: This module needs paramiko to automatically write backups to the FTP through SFTP. Please install paramiko on your system. (sudo pip3 install paramiko)

Note ; I just installed the module paramiko but i can't install with sudo

**Comment by prajithpipt** [comment]
@Yenthe666 
I have an error during the installation of auto_backup in Odoo 12

raise Exception(_('Module loading %s failed: file %s could not be processed:\n %s') % (module, fname, warning_msg))
Exception: Module loading auto_backup failed: file auto_backup/security/ir.model.access.csv could not be processed:
 No matching record found for external id 'model_db_backup' in field 'Object'
No matching record found for external id 'model_db_backup' in field 'Object'
Missing required value for the field 'Object' (model_id)
Missing required value for the field 'Object' (model_id)
